docs/running-clusterfuzzlite/gitlab.md

```diff
@@ -0,0 +1,229 @@
+      variables:
+        MODE: "code-change"
+  before_script:
+    # gitlab's container name is not its hostname !
+    - export CFL_CONTAINER_ID=`cut -c9- < /proc/1/cpuset`
+  script:
+    # will build and run the fuzzers
+    - python3 "/opt/oss-fuzz/infra/cifuzz/cifuzz_combined_entrypoint.py"
+  artifacts:
+    # upload artifacts when a crash makes the job fail
+    when: always
+    paths:
+      - artifacts/crashes/
+```
+{% endraw %}
+
+You may also wish to set [tags](https://docs.gitlab.com/runner/#tags) to select a relevant runner.
+
+Optionally, edit the following variables to customize your settings:
+- `SANITIZER` Change or enable more sanitizers.
+- `LANGUAGE` Define the language of your project.
+- `CFL_BRANCH` Branch to fuzz, default is `CI_DEFAULT_BRANCH`.
+- `FILESTORE` To use corpus produced by other jobs.
+- `FUZZ_SECONDS` Change the amount of time spent fuzzing.
+- `CFL_ARTIFACTS_DIR` : To save your artifacts in a different directory than `artifacts`
+
+For `SANITIZER`, you may also use a matrix to use multiple sanitizers with the same job.
+
+### Batch fuzzing and corpus pruning
+
+Batch fuzzing enables continuous, regular fuzzing on your latest HEAD and
+allows a corpus of inputs to build up over time, which greatly improves the
+effectiveness of fuzzing. Batch fuzzing should be run on a schedule.
+
+To enable batch fuzzing, add the following to
+`.gitlab-ci.yml`:
+
+{% raw %}
+```yaml
+# this name is hardcoded and cannot be changed for gitlab artifacts filestore
+clusterfuzzlite-corpus:
+  image:
+    name: google/clusterfuzzlite/actions/build_fuzzers@v1
+    entrypoint: [""]
+  stage: fuzz
+  rules:
+    - if: $MODE == "prune"
+    - if: $MODE == "batch"
+  before_script:
+    # gitlab's container name is not its hostname !
+    - export CFL_CONTAINER_ID=`cut -c9- < /proc/1/cpuset`
+  script:
+    - python3 "/opt/oss-fuzz/infra/cifuzz/cifuzz_combined_entrypoint.py"
+  artifacts:
+    when: always
+    paths:
+      - artifacts/corpus/
+      - artifacts/crashes/
+```
+{% endraw %}
+
+You should then define two [schedules](https://docs.gitlab.com/ee/ci/pipelines/schedules.html)
+In one, you should set the variable `MODE` to `batch` to run the actual batch fuzzing.
+In the other, you should set the variable `MODE` to `prune` for corpus pruning once a day.
+These schedules should target the main/default/`CFL_BRANCH` branch.
+
+### Continuous builds
+
+The continuous build task causes a build to be triggered and uploaded
+whenever a new push is done to main/default branch.
+
+Continuous builds are used when a crash is found during PR fuzzing to determine whether the crash was newly introduced.
+If the crash is not novel, PR fuzzing will not report it.
+This means that there will be fewer unrelated failures when running code change
+fuzzing.
+
+To set up continuous builds, add the following to `.gitlab-ci.yml`:
+
+{% raw %}
+```yaml
+# this name is hardcoded and cannot be changed for gitlab artifacts filestore
```

Comment:
This is only needed with the gitlab artifacts filestore.

When ClusterFuzzLite Gitlab's artifacts filestore uses Gitlab API, it looks for a job with this specific name `clusterfuzzlite-corpus`
Maybe we could make it configurable instead of hardcoded.

Reply:
Hardcoding job names seems not very intuitive. Developers use any sorts of job names. If necessary, at least it should be configurable.

Reply:
jobs names are no longer required to be hard coded